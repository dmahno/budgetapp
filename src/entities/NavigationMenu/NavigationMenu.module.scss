.navigation {
  list-style-type: none;
  display: flex;
  gap: $indent-l;
}
