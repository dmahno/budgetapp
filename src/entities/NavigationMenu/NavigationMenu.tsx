import {menuLinks} from './model/data/menuLinks';
import {MenuItem} from './ui/MenuItem/MenuItem';
import styles from './NavigationMenu.module.scss';

export const NavigationMenu = () => {
  return (
    <nav>
      <ul className={styles.navigation}>
        {menuLinks.map((item) => {
          const {url, label} = item;
          return <MenuItem key={label} url={url} label={label} />;
        })}
      </ul>
    </nav>
  );
};
