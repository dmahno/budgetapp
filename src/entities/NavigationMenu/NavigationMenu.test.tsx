import {render, screen} from '@testing-library/react';
import {describe, it, expect, vi} from 'vitest';
import {MemoryRouter} from 'react-router-dom';

import {NavigationMenu} from './NavigationMenu';

vi.mock('./model/data/menuLinks', () => ({
  menuLinks: [
    {url: '/home', label: 'Home'},
    {url: '/about', label: 'About'},
    {url: '/contact', label: 'Contact'},
  ],
}));

describe('NavigationMenu Component', () => {
  it('renders the correct number of menu items', () => {
    render(
      <MemoryRouter>
        <NavigationMenu />
      </MemoryRouter>,
    );
    const menuItems = screen.getAllByRole('link');
    expect(menuItems).toHaveLength(3);
  });

  it('renders each menu item with correct label and href', () => {
    render(
      <MemoryRouter>
        <NavigationMenu />
      </MemoryRouter>,
    );
    const menuLinks = [
      {label: 'Home', href: '/home'},
      {label: 'About', href: '/about'},
      {label: 'Contact', href: '/contact'},
    ];

    menuLinks.forEach(({label, href}) => {
      const linkElement = screen.getByRole('link', {name: label});
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveAttribute('href', href);
    });
  });

  it('matches the snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <NavigationMenu />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
