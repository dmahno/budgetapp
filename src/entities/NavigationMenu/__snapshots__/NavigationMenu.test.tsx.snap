// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu Component > matches the snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="_navigation_7d6e8c"
    >
      <li>
        <a
          class="_link_9fe8fa"
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="_link_9fe8fa"
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          class="_link_9fe8fa"
          href="/contact"
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
